### DISCUSSION TITLE
sentry.io is blocked: ERR_BLOCKED_BY_CLIENT

### DISCUSSION THREAD

**Discussion by mokshadharma** [qutebrowser/qutebrowser#7585] [Q&A] [answered]
When I try to log in to [Sentry](https://sentry.io/welcome/) via [SAML](https://docs.sentry.io/product/accounts/sso/), I get an error:

```
sentry.io is blocked
Requests to the server have been blocked by an extension.
Try disabling your extensions.
ERR_BLOCKED_BY_CLIENT
```

(Please see the attached screenshot below)

I'm using qutebrowser v2.5.2, on macOS Ventura 13.2.1

I'm able to log in to Sentry using Firefox 109.0.1 and Google Chrome 110.0.5481.77 (Official Build) (arm64) just fine.

<img width="455" alt="sentry-io-is-blocked" src="https://user-images.githubusercontent.com/118926161/218584550-a40bde34-4d69-433b-bcfa-a0602c9203f9.png">

**Comment by The-Compiler** [comment, marked as answer]
This means it's blocked by something on your hosts/adblock lists (depending on `content.blocking.method`). Try an `:adblock-update`, and if that doesn't help, review your filter lists, or add it to `content.blocking.whitelist`.

**Reply by mokshadharma** [reply]
Thanks, but `:adblock-update` didn't help.

I then tried putting this in my `config.py`:

```python
c.content.blocking.whitelist = ['sentry.io', 'mycompany.sentry.io', 'mycompany.sentry.io/*']
```

Then did a `:config-source` within qutebrowser and reloaded the page.  That got me past the above error message, only to see it go to https://mycompany.sentry.io/issues and show me an animation of the Sentry logo, with a message saying:

```
Please wait while we load an obnoxious amount of JavaScript.

You may need to disable adblocking extensions to load Sentry.
```

and that just hangs there forever.

I also tried:

```python
with config.pattern('*://mycompany.sentry.io/') as p:
    p.content.blocking.enabled = False

with config.pattern('*://mycompany.sentry.io/*') as p:
    p.content.blocking.enabled = False
```

and typed `:config-source` inside qutebrowser and reloaded the page, but that didn't help.

Anything else I could try?

**Reply by toofar** [reply]
Have a look in the network tab in the web inspector and see what's getting blocked. The other other domain I could see is `s1.sentry-cdn.com`.

**Comment by mokshadharma** [comment]
I tried putting this in to my `config.py`:

```python
with config.pattern('*://sentry.io') as p:
    p.content.blocking.enabled = False

with config.pattern('*://sentry.io/*') as p:
    p.content.blocking.enabled = False

with config.pattern('*://s1.sentry-cdn.com') as p:
    p.content.blocking.enabled = False

with config.pattern('*://sentry-cdn.com') as p:
    p.content.blocking.enabled = False

with config.pattern('*://sentry-cdn.com/*') as p:
    p.content.blocking.enabled = False

with config.pattern('*://*.sentry-cdn.com/*') as p:
    p.content.blocking.enabled = False

with config.pattern('*://s1.sentry-cdn.com/*') as p:
    p.content.blocking.enabled = False

with config.pattern('*://mycompany.sentry.io') as p:
    p.content.blocking.enabled = False

with config.pattern('*://mycompany.sentry.io/*') as p:
    p.content.blocking.enabled = False


c.content.blocking.whitelist = ['sentry.io', 'mycompany.sentry.io', 'https://mycompany.sentry.io/issues', 'sentry-cdn.com', 's1.sentry-cdn.com', '*.sentry-cdn.com']
```

But it still did not help.  The page is stuck on that same Sentry logo animation, with the same message as above.

I also tried selecting the `Blocked Requests` checkbox in the `Network` tab of the `:devtools` tab (see screenshot below), and it did not show any blocked requests.

<img width="1918" alt="sentry-and-devtools" src="https://user-images.githubusercontent.com/118926161/219089234-7513d2e5-c199-4f29-a387-a8fc23e37796.png">

**Reply by toofar** [reply]
"blocked" requests might show up as "cancelled" or something, try removing the filter and reviewing anything that shows in red.

**Reply by mokshadharma** [reply]
There's only one thin vertical line, just as you see in your screenshot, and there's nothing that says "cancelled" or anything like that.  All have 200 as the Status.  I've added every single domain in all of the requests as the following pattern (substituting each domain for `example.com` below):

```python
with config.pattern('*://example.com') as p:
    p.content.blocking.enabled = False

with config.pattern('*://example.com/*') as p:
    p.content.blocking.enabled = False

with config.pattern('*://*.example.com') as p:
    p.content.blocking.enabled = False

with config.pattern('*://*.example.com/*') as p:
    p.content.blocking.enabled = False

c.content.blocking.whitelist = ['https://example.com', 'https://example.com/*', 'https://*.example.com', 'https://*.example.com/*']
```

Then I did a `:config-source` and reloaded the page, but am still stuck on that animated Sentry logo.

**Reply by toofar** [reply]
Well at least the ERR_BLOCKED_BY_CLIENT error is gone, right!
Make sure you have this little guy unchecked 
![image](https://user-images.githubusercontent.com/7419144/220829255-6c96b66b-8462-42d8-8453-6d7aa622d696.png)

Other than that try checking the console tab of the developer tools and see if there are any interesting errors.

**Reply by mokshadharma** [reply]
Yes, the "Blocked Requests" checkbox is unchecked.

The following screenshot shows what I saw when looking in the console tab of the developer tools.  Not that it's likely to matter, but I've redacted the hashes from the screenshot.

<img width="960" alt="Screenshot 2023-02-24 at 10 32 03 AM" src="https://user-images.githubusercontent.com/118926161/221220292-87035824-2d4c-4663-9d05-cbb538f70d93.png">

When I hover over the syntax errors, I see they're all from scripts on `s1.sentry-cdn.com`.

The `Unrecognized feature: 'payment'` is from `js.stripe.com`

**Reply by toofar** [reply]
The payment and inline style errors shouldn't be the problem, they are "normal".

Hmm, that smells like an old-webengine new-js-feature issue. What webengine version are you on (see :version), I just tried with `QtWebEngine 5.15.12` and it seems to work fine (although I'm logging in via GitHub, which is probably OAuth).

Although the internet also says it could be an ad blocking issue again: https://help.looker.com/hc/en-us/articles/4420194545171-What-does-a-ChunkLoadError-in-the-JS-console-mean-Community-

You could also try with `--temp-basedir` and see if it works in a fresh instance.

**Reply by mokshadharma** [reply]
`:version` shows:

`PyQt5.QtWebEngine: 5.15.5`

**Reply by The-Compiler** [reply]
Yeah, that does indeed seem like a problem with an old QtWebEngine and Sentry requiring more modern JS syntax - in other words, probalby not related to the blocking at all.

Could you try the Qt 6 branch (see #7202), and check if that works better?

**Reply by mokshadharma** [reply]
I'm currently limited to installing with homebrew, so can't try that branch until homebrew has it.

**Comment by jaimefps** [comment]
As @The-Compiler pointed out at the start, ad-blockers can mess with Sentry. In my case it was my using the Brave browser (which blocks a lot of stuff by default) that gave me issues. Worked fine when testing from a Google Chrome browser.
